File: android/src/main/java/com/cachevideo/CacheVideoHttpProxyPackage.java
package com.cachevideo;

import android.util.Log;

import androidx.annotation.Nullable;

import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.module.model.ReactModuleInfo;
import com.facebook.react.module.model.ReactModuleInfoProvider;
import com.facebook.react.TurboReactPackage;

import java.util.HashMap;
import java.util.Map;

public class CacheVideoHttpProxyPackage extends TurboReactPackage {

  @Nullable
  @Override
  public NativeModule getModule(String name, ReactApplicationContext reactContext) {
    if (name.equals(CacheVideoHttpProxyModule.NAME)) {
      Log.d(CacheVideoHttpProxyModule.NAME, reactContext.toString());
      return new CacheVideoHttpProxyModule(reactContext);
    } else {
      return null;
    }
  }

  @Override
  public ReactModuleInfoProvider getReactModuleInfoProvider() {
    return () -> {
      final Map<String, ReactModuleInfo> moduleInfos = new HashMap<>();
      boolean isTurboModule = BuildConfig.IS_NEW_ARCHITECTURE_ENABLED;
      moduleInfos.put(
              CacheVideoHttpProxyModule.NAME,
              new ReactModuleInfo(
                      CacheVideoHttpProxyModule.NAME,
                      CacheVideoHttpProxyModule.NAME,
                      false, // canOverrideExistingModule
                      false, // needsEagerInit
                      true, // hasConstants
                      false, // isCxxModule
                      isTurboModule // isTurboModule
      ));
      return moduleInfos;
    };
  }
}

File: android/src/main/java/com/cachevideo/CacheVideoHttpProxyPackage.kt
package com.cachevideo

import com.facebook.react.BaseReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.model.ReactModuleInfo
import com.facebook.react.module.model.ReactModuleInfoProvider

class CacheVideoHttpProxyPackage : BaseReactPackage() {

  override fun getModule(name: String, reactContext: ReactApplicationContext): NativeModule? {
    return if (name == CacheVideoHttpProxyModule.NAME) {
      CacheVideoHttpProxyModule(reactContext)
    } else {
      null
    }
  }

  override fun getReactModuleInfoProvider(): ReactModuleInfoProvider {
    return ReactModuleInfoProvider {
      mapOf(
        CacheVideoHttpProxyModule.NAME to ReactModuleInfo(
          CacheVideoHttpProxyModule.NAME,
          CacheVideoHttpProxyModule.NAME,
          false, // canOverrideExistingModule
          false, // needsEagerInit
          false, // isCxxModule
          true // isTurboModule
        )
      )
    }
  }
}
